tests/scene/test_packed_scene.h

```diff
@@ -56,6 +56,67 @@ TEST_CASE("[PackedScene] Pack Scene and Retrieve State") {
 	memdelete(scene);
 }
 
+TEST_CASE("[PackedScene] Test That Correct Signals are Preserved when Packing Scene") {
+	// Create main scene
+	// root
+	// `- sub_node (local)
+	// `- sub_scene (instance of another scene)
+	//    `- sub_scene_node (owned by sub_scene)
+	Node *main_scene_root = memnew(Node);
+	Node *sub_node = memnew(Node);
+	Node *sub_scene_root = memnew(Node);
+	Node *sub_scene_node = memnew(Node);
+
+	main_scene_root->add_child(sub_node);
+	sub_node->set_owner(main_scene_root);
+
+	sub_scene_root->add_child(sub_scene_node);
+	sub_scene_node->set_owner(sub_scene_root);
+
+	main_scene_root->add_child(sub_scene_root);
+	sub_scene_root->set_owner(main_scene_root);
+
+	SUBCASE("Signals that should be saved") {
+		int main_flags = Object::CONNECT_PERSIST;
+		// sub node to a node in main scene
+		sub_node->connect("ready", callable_mp(main_scene_root, &Node::is_ready), main_flags);
+		// subscene root to a node in main scene
+		sub_scene_root->connect("ready", callable_mp(main_scene_root, &Node::is_ready), main_flags);
+		//subscene root to subscene root (connected within main scene)
+		sub_scene_root->connect("ready", callable_mp(sub_scene_root, &Node::is_ready), main_flags);
+
+		// Pack the scene.
+		PackedScene packed_scene;
+		const Error err = packed_scene.pack(main_scene_root);
+		CHECK(err == OK);
+
+		// Make sure the right connections are in packed scene
+		Ref<SceneState> state = packed_scene.get_state();
```

Comment:
```suggestion
		Ref<PackedScene> packed_scene;
		packed_scene.instantiate();
		const Error err = packed_scene->pack(main_scene_root);
		CHECK(err == OK);

		// Make sure the right connections are in packed scene.
		Ref<SceneState> state = packed_scene->get_state();
```
Use `Ref` for RefCounted for correctness